src/engine/Server.cpp

```diff
@@ -425,6 +427,29 @@ string Server::composeResponseJson(const string& query,
   return os.str();
 }
 
+// _____________________________________________________________________________
+string Server::composeResponseJson(const string& query,
```

Comment:
Since both `std::runtime_error` and `ParseException` are accessed via `what()` I would either template this on the exception type or let it take a string and do that outside. Either way this looks like duplication of (already ugly) code.

Reply:
I switched it over to catching and handling `std::exception`.